classes/module/Module.php

```diff
@@ -3144,6 +3144,23 @@ protected function trans($id, array $parameters = array(), $domain = null, $loca
         $parameters['legacy'] = 'htmlspecialchars';
         return $this->getTranslator()->trans($id, $parameters, $domain, $locale);
     }
+
+    /**
+     * Check if the module uses the new translation system
+     * @return bool
+     */
+    public function isUsingNewTranslationsSystem()
+    {
+        $moduleName = $this->name;
+        $domains = array_keys($this->context->getTranslator()->getCatalogue()->all());
+        $moduleName = preg_replace('/^ps_(\w+)/', '$1', $moduleName);
+
+        if (count(preg_grep('/'.$moduleName.'/i', $domains))) {
```

Comment:
There are several problems with that:
1) I don't understand the need for "#"
2) You are merging all domains needlessly (you only need to check them until you find one that matches)
3) Merging them all without a delimiter might create false-positives.

You can simply do it like this:
```php
foreach ($domains as $domain) {
    if (false !== stripos($domain, $moduleName))) {
        return true;
    }
}
```